internal/controller/dataplane/openstackdataplanenodeset_controller.go

```
@@ -517,18 +522,23 @@ func checkDeployment(ctx context.Context, helper *helper.Helper,
+	// Sort relevant deployments from oldest to newest, then take the last one.
+	// The latest transitioned deployment is what last acted on the node;
+	// tie-break on creationTimestamp then name for determinism.
 	var latestRelevantDeployment *dataplanev1.OpenStackDataPlaneDeployment
 	if len(relevantDeployments) > 0 {
 		slices.SortFunc(relevantDeployments, func(a, b *dataplanev1.OpenStackDataPlaneDeployment) int {
 			aReady := a.Status.Conditions.Get(condition.DeploymentReadyCondition)
 			bReady := b.Status.Conditions.Get(condition.DeploymentReadyCondition)
 			if aReady != nil && bReady != nil {
-				if aReady.LastTransitionTime.Before(&bReady.LastTransitionTime) {
-					return -1
+				if c := aReady.LastTransitionTime.Time.Compare(bReady.LastTransitionTime.Time); c != 0 {
```

golangci-lint (2.13.2)

[error] QF1008: could remove embedded field "Time" from selector

(staticcheck)